int32 period

int32 port

int8 subnetwork